appium_helpers.py

```diff
@@ -0,0 +1,160 @@
+        return driver.find_element_by_ios_uiautomation(s)
+
+    @staticmethod
+    def find_and_wait_button(driver, name, wait_time=5):
+        s = '.elements()["{}"]'.format(name)
+        return iOS.wait_by(driver,
+                           EC.presence_of_element_located,
+                           By.IOS_UIAUTOMATION,
+                           s,
+                           wait_time)
+
+    @staticmethod
+    def find_and_wait_text_from_webview(driver, name, wait_time=5):
+        s = '.scrollViews()[0].webViews()[0].staticTexts()["{}"]'.format(name)
+        return iOS.wait_by(driver,
+                           EC.presence_of_element_located,
+                           By.IOS_UIAUTOMATION,
+                           s,
+                           wait_time)
+
+    @staticmethod
+    def find_all_webview_elements_by_xpath(driver):
+        return driver.find_elements_by_xpath('//UIAApplication[1]/UIAWindow[1]/UIAScrollView[1]/UIAWebView[1]/*')
+
+    @staticmethod
+    def find_element_by_xpath(driver, xpath):
+        return driver.find_element_by_xpath(xpath)
+
+    @staticmethod
+    def find_webview_elements_by_xpath(driver, xpath):
+        return driver.find_elements_by_xpath('//UIAApplication[1]/UIAWindow[1]/UIAScrollView[1]/UIAWebView[1]/' + xpath)
+
+    @staticmethod
+    def find_all_elements_by_xpath(driver):
+        return driver.find_elements_by_xpath('//*[not(*)]')
+
+    @staticmethod
+    def find_all_window_elements_by_xpath(driver):
+        return driver.find_elements_by_xpath('//UIAApplication[1]/UIAWindow[1]/*')
+
+    @staticmethod
+    def find_all_window_elements(driver):
+        return driver.find_elements_by_ios_uiautomation('.elements()')
+
+    @staticmethod
+    def wait_by(driver, expected_condition, by_method, element_identifier, wait_time=5):
+        """
+        See: http://selenium-python.readthedocs.io/waits.html
+        Args:
+            driver: instance of Appium.webdriver
+            expected_condition: what we condition we expect to happen
+            by_method: by with what Appium.webdriver.method we try to find the element
+            element_identifier: depending on 'by_method' what indentifier we try to use to locate the elem
+            wait_time: time to wait element to appear
+
+        Returns: instance of appium.webdriver.webelement or TimeoutException
+        """
+        return WebDriverWait(driver, wait_time).until(expected_condition((by_method, element_identifier)))
+
+
+
+    @staticmethod
+    def toggle_orientation(driver):
+        current_orientation = driver.orientation
+        if current_orientation == "PORTRAIT":
+            driver.orientation = "LANDSCAPE"
+        else:
+            driver.orientation = "PORTRAIT"
+
+    @staticmethod
+    def get_xml_tree(driver):
+        return driver.execute_script('au.mainApp().getTreeForXML()')
+
+    @staticmethod
+    def screenshot_fast(driver, name, dir=getcwd()+'/screenshots'):
```

Comment:
well it's usually fine to use non-safe functions in safe environment. I mean dir can be created before method call.